src/components/tree-select.tsx

```diff
@@ -56,19 +59,23 @@ const TreeSelect = <Item extends BasicItem<Item>>({
 }: Except<DropdownButtonProps, 'children' | 'label'> &
   TreeSelectProps<Item>) => {
   const [activeNodes, setActiveNodes] = useState(defaultActiveNodes);
-  const [openNodes, setOpenNodes] = useState<BasicItem<Item>['id'][]>([]);
+
+  const [openNodes, setOpenNodes] = useState<BasicItem<Item>['id'][]>(
+    Array.from(getSingleChildren(data))
+  );
   const [autocompleteShowDropdown, setAutocompleteShowDropdown] =
     useState(false);
 
   const toggleNode = useCallback(
-    (node: AutocompleteItemType | BasicItem<Item>) => {
-      if (openNodes.includes(node.id)) {
-        setOpenNodes(openNodes.slice(0, openNodes.indexOf(node.id)));
-      } else {
-        setOpenNodes([...openNodes, node.id]);
-      }
-    },
-    [openNodes]
+    (node: AutocompleteItemType | BasicItem<Item>) =>
+      setOpenNodes((openNodes) => {
+        const index = openNodes.indexOf(node.id);
+        if (index === -1) {
+          return [...openNodes, node.id];
+        }
+        return openNodes.filter((id) => id !== node.id);
+      }),
```

Comment:
```suggestion
      setOpenNodes((openNodes) =>
        openNodes.includes(node.id)
          ? openNodes.filter((id) => id !== node.id)
          : [...openNodes, node.id]
      ),
```
Just suggesting because I initially thought something is done with `index` other than just checking inclusion

Reply:
Something was being done with the index, because I was planning on replacing the previous `slice` by a `splice`, but then I switched to filter because I though it was easier to understand the meaning behind what was being done and forgot to remove that index.
I'll take your suggested change.